src/ucp/proto/proto_common.inl

```diff
@@ -357,6 +357,16 @@ ucp_proto_request_pack_rkey(ucp_request_t *req, ucp_md_map_t md_map,
                 ucp_datatype_class_names[dt_iter->dt_class]);
 
     memh = dt_iter->type.contig.memh;
+
+    /* Since global VA registration doesn't support invalidation yet, and error
+     * handling is enabled on this EP, we replace GVA registrations with
+     * regular ones */
+    if ((memh->flags & UCP_MEMH_FLAG_HAS_GVA) &&
+        ucp_ep_config_err_mode_eq(req->send.ep,
+                                  UCP_ERR_HANDLING_MODE_PEER)) {
```

Comment:
for optimization:
```suggestion
    if (ucp_ep_config_err_mode_eq(req->send.ep,
                                  UCP_ERR_HANDLING_MODE_PEER) &&
        ucs_unlikely(memh->flags & UCP_MEMH_FLAG_HAS_GVA)) {
```

Reply:
AFAIK unlikely doesn't work for subexpressions, only for entire if condition.
do you expect checking ep configuration to be faster then memh flag? i thought vice-versa

Reply:
> AFAIK unlikely doesn't work for subexpressions, only for entire if condition.

Do you have a doc ref for this? I could not find anything official about this..

> do you expect checking ep configuration to be faster then memh flag? i thought vice-versa

In the most performance-important case (no err handling) - we would have GVA, and the suggested order will have only one branch (since first check fails).